tests/functional/history/test_db.py

```diff
@@ -301,3 +299,168 @@ def test_can_read_most_recent_records(self):
         records = set([record['event_type'] for record
                        in reader.iter_latest_records()])
         self.assertEqual(set(['foo', 'bar']), records)
+
+
+class TestDatabaseHistoryHandler(unittest.TestCase):
+    UUID_PATTERN = re.compile(
+        '^[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}$',
+        re.I
+    )
+
+    def setUp(self):
+        self.db = DatabaseConnection(':memory:')
+        self.writer = DatabaseRecordWriter(connection=self.db)
+        self.handler = DatabaseHistoryHandler(writer=self.writer)
+
+    def _assert_last_record_structure(self, event_type, payload, source,
```

Comment:
It's kind of hard to follow this. I honestly would prefer there be the following:

* `_get_last_record()` to get the last record and the payload already loaded.
* `_assert_record_is_correct_structure()` which does the assertion on the tuple
* Then have more helper assertion methods like `_assert_expected_event_type()`, `_assert_expected_source_type()`, and `_assert_expected_payload()`, `_assert_has_request_id()`

The problem now for me is the assertion is too vague as it is now and makes it hard to follow the assertions.